src/Module/Conversation/Channel.php

```diff
@@ -0,0 +1,398 @@
+			} else {
+				$condition = ["(`comments` >= ? OR `activities` >= ?) AND `contact-type` != ?", self::getMedianComments(4), self::getMedianActivities(4), Contact::TYPE_COMMUNITY];
+			}
+		} elseif (self::$content == self::FORYOU) {
+			$cid = Contact::getPublicIdByUserId(DI::userSession()->getLocalUserId());
+
+			$condition = ["(`owner-id` IN (SELECT `relation-cid` FROM `contact-relation` WHERE `cid` = ? AND `thread-score` > ?) OR
+				((`comments` >= ? OR `activities` >= ?) AND `owner-id` IN (SELECT `pid` FROM `account-user-view` WHERE `uid` = ? AND `rel` IN (?, ?))) OR
+				( `owner-id` IN (SELECT `pid` FROM `account-user-view` WHERE `uid` = ? AND `rel` IN (?, ?) AND `notify_new_posts`)))",
+				$cid, self::getMedianThreadScore($cid, 4), self::getMedianComments(4), self::getMedianActivities(4), DI::userSession()->getLocalUserId(), Contact::FRIEND, Contact::SHARING,
+				DI::userSession()->getLocalUserId(), Contact::FRIEND, Contact::SHARING];
+		} elseif (self::$content == self::FOLLOWERS) {
+			$condition = ["`owner-id` IN (SELECT `pid` FROM `account-user-view` WHERE `uid` = ? AND `rel` = ?)", DI::userSession()->getLocalUserId(), Contact::FOLLOWER];
+		} elseif (self::$content == self::IMAGE) {
+			$condition = ["`media-type` & ?", 1];
+		} elseif (self::$content == self::VIDEO) {
+			$condition = ["`media-type` & ?", 2];
+		} elseif (self::$content == self::AUDIO) {
+			$condition = ["`media-type` & ?", 4];
+		}
+
+		$condition[0] .= " AND NOT EXISTS(SELECT `cid` FROM `user-contact` WHERE `uid` = ? AND `cid` = `post-engagement`.`owner-id` AND (`ignored` OR `blocked` OR `collapsed`))";
+		$condition[] = DI::userSession()->getLocalUserId();
+
+		if ((self::$content != self::WHATSHOT) && !is_null(self::$accountType)) {
+			$condition[0] .= " AND `contact-type` = ?";
+			$condition[] = self::$accountType;
+		}
+
+		$params = ['order' => ['created' => true], 'limit' => self::$itemsPerPage];
+
+		if (!empty(self::$item_id)) {
+			$condition[0] .= " AND `uri-id` = ?";
+			$condition[] = self::$item_id;
+		} else {
+			if (!empty($_REQUEST['no_sharer'])) {
+				$condition[0] .= " AND NOT `uri-id` IN (SELECT `uri-id` FROM `post-user` WHERE `post-user`.`uid` = ? AND `post-user`.`uri-id` = `post-engagement`.`uri-id`)";
+				$condition[] = DI::userSession()->getLocalUserId();
+			}
+
+			if (isset(self::$max_id)) {
+				$condition[0] .= " AND `created` < ?";
+				$condition[] = self::$max_id;
+			}
+
+			if (isset(self::$min_id)) {
+				$condition[0] .= " AND `created` > ?";
+				$condition[] = self::$min_id;
+
+				// Previous page case: we want the items closest to min_id but for that we need to reverse the query order
+				if (!isset(self::$max_id)) {
+					$params['order']['created'] = false;
+				}
+			}
+		}
+
+		$items = DBA::selectToArray('post-engagement', ['uri-id', 'created'], $condition, $params);
```

Comment:
Please use the `Database` object you can add to this class' constructor and save to a class property to perform `DBA` methods.

Reply:
done